webpack.config.js

```diff
@@ -1,65 +1,65 @@
-const path = require('path');
 const {HotModuleReplacementPlugin} = require('webpack');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 const port = 3000;
+const context = __dirname + '/src';
+const environment = process.env.NODE_ENV;
 
 const webpackConfig = {
   name: 'client',
   target: 'web',
+  context,
   entry: {
-    app: [
-      './src/index.tsx',
-      `webpack-dev-server/client?http://localhost:${port}/`,
-      'webpack/hot/dev-server'
-    ],
+    app: './index.tsx'
   },
   output: {
     filename: '[name].js',
-    path: 'dist',
-    publicPath: ''
+    path: __dirname + 'dist',
   },
   resolve: {
-    extensions: ["", ".ts", ".tsx", ".js", "jsx"]
+    extensions: ['.ts', '.tsx', '.js', 'jsx']
   },
   devtool: 'source-map',
   module: {
-    preLoaders: [{
+    rules: [{
+      enforce: 'pre',
       test: /\.tsx?$/,
-      loader: "tslint",
-      exclude: /node_modules/
-    }],
-    loaders: [{
+      use: 'tslint-loader',
+      exclude: /node_modules/,
+      options: {
+        configFile: './tslint.json',
+        emitErrors: true,
+        failOnHint: true
+      }
+    },
+    {
       test: /\.tsx?$/,
-      loader: "awesome-typescript",
+      use: 'awesome-typescript-loader',
       exclude: /node_modules/
     }]
   },
   plugins: [
     new HtmlWebpackPlugin({
-      template: './src/index.html',
+      template: './index.html',
       hash: true,
       filename: 'index.html',
       inject: 'body',
     }),
     new HotModuleReplacementPlugin()
-  ],
-  tslint: {
-    configFile: './tslint.json',
-    emitErrors: true,
-    failOnHint: true
-  },
-  devServer: {
+  ]
+};
+
+if (environment === 'dev') {
```

Comment:
like this 
https://github.com/ngParty/React-scaffold/blob/develop/webpack.config.js#L53